new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainOnPV.java

```diff
@@ -0,0 +1,682 @@
+    assertTrue(secretCreated, String.format("create secret failed for %s", wlSecretName));
+  }
+
+  /**
+   * Create a persistent volume and persistent volume claim.
+   * @throws IOException when creating pv path fails
+   */
+  private void createPVandPVC() throws IOException {
+    logger.info("creating persistent volume and persistent volume claim");
+
+    Path pvHostPath = Files.createDirectories(Paths.get(
+        PV_ROOT, this.getClass().getSimpleName(), domainUid + "-persistentVolume"));
+    logger.info("Creating PV directory {0}", pvHostPath);
+    FileUtils.deleteDirectory(pvHostPath.toFile());
+    Files.createDirectories(pvHostPath);
+    V1PersistentVolume v1pv = new V1PersistentVolume()
+        .spec(new V1PersistentVolumeSpec()
+            .addAccessModesItem("ReadWriteMany")
+            .storageClassName(domainUid + "-weblogic-domain-storage-class")
+            .volumeMode("Filesystem")
+            .putCapacityItem("storage", Quantity.fromString("5Gi"))
+            .persistentVolumeReclaimPolicy("Recycle")
+            .accessModes(Arrays.asList("ReadWriteMany"))
+            .hostPath(new V1HostPathVolumeSource()
+                .path(pvHostPath.toString())))
+        .metadata(new V1ObjectMetaBuilder()
+            .withName(pvName)
+            .withNamespace(domainNamespace)
+            .build()
+            .putLabelsItem("weblogic.resourceVersion", "domain-v2")
+            .putLabelsItem("weblogic.domainUid", domainUid));
+    boolean success = assertDoesNotThrow(
+        () -> createPersistentVolume(v1pv),
+        "Persistent volume creation failed, "
+        + "look at the above console log messages for failure reason in ApiException responsebody"
+    );
+    assertTrue(success, "PersistentVolume creation failed");
+
+    V1PersistentVolumeClaim v1pvc = new V1PersistentVolumeClaim()
+        .spec(new V1PersistentVolumeClaimSpec()
+            .addAccessModesItem("ReadWriteMany")
+            .storageClassName(domainUid + "-weblogic-domain-storage-class")
+            .volumeName(pvName)
+            .resources(new V1ResourceRequirements()
+                .putRequestsItem("storage", Quantity.fromString("5Gi"))))
+        .metadata(new V1ObjectMetaBuilder()
+            .withName(pvcName)
+            .withNamespace(domainNamespace)
+            .build()
+            .putLabelsItem("weblogic.resourceVersion", "domain-v2")
+            .putLabelsItem("weblogic.domainUid", domainUid));
+
+    success = assertDoesNotThrow(
+        () -> createPersistentVolumeClaim(v1pvc),
+        "Persistent volume claim creation failed, "
+        + "look at the above console log messages for failure reason in ApiException responsebody"
+    );
+    assertTrue(success, "PersistentVolumeClaim creation failed");
+  }
+
+
+  /**
+   * Install WebLogic operator and wait until the operator pod is ready.
+   */
+  private static void installAndVerifyOperator() {
```

Comment:
i think we have one of these somewhere else - we should consider refactoring to avoid duplication

Reply:
it is done in a separate PR for all classes